release-versions/generate_release_file.py

```diff
@@ -0,0 +1,151 @@
+from quik import FileLoader
+from collections import ChainMap
+import requests
+import os
+from datetime import datetime
+
+releases = []
+
+headers = {'Content-Type': 'application/json',
+           'Authorization': f'Bearer {os.environ["GITHUB_API_TOKEN"]}'}
+
+
+def url(repo, suffix):
+    return f"https://api.github.com/repos/nationalarchives/{repo}/{suffix}"
+
+
+def get_versions(repository_name):
+    branches = requests.get(url(repository_name, "branches"), headers=headers).json()
+    if "message" not in branches:
```

Comment:
What does this filter do?

Reply:
I've no idea :smile: It doesn't do anything, there's no `message` parameter in the branches json and if I take it out, it still works. I've taken it out.

Reply:
Where did it come from originally? Was this script based on some other example that used `message`?

Reply:
We do actually need this filter. This is what made the build fail this morning.
The Jenkins Github user doesn't have access to `tdr-configurations` I guess this is a problem with the token and it being a private repo but when you call the `/branches` endpoint for that repo, you get `'{"message": "Not Found", "documentation_url": "https://docs.github.com/rest/reference/repos#list-branches"}'`
You don't get the `message` parameter in a successful call so we're just filtering out any failed calls.

Reply:
Maybe we should update the GitHub API key to allow calls to private repos? I think we'll need to do that anyway to get the PR monitor to include them.

Reply:
We are using the Github API key, it just doesn't have permission to call private repos. I think it's a separate permission on the token but I don't have access to it to check.

Reply:
If it's just a case of adding the permission, then I can do that. Do you want me to try it?

Reply:
Yeah, please. There's a setting to allow access to private repos I think.

Reply:
That's worked so I've taken the filter out.